libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtils.kt

```diff
@@ -539,9 +539,9 @@ internal object MapboxRouteLineUtils {
      */
     internal val extractRouteDataWithTrafficAndRoadClassDeDuped: (
         route: DirectionsRoute,
-        trafficCongestionProvider: (RouteLeg) -> List<String>?
+        trafficCongestionProvider: TrafficCongestionProvider
     ) -> List<ExtractedRouteData> =
-        { route: DirectionsRoute, trafficCongestionProvider: (RouteLeg) -> List<String>? ->
+        { route: DirectionsRoute, trafficCongestionProvider: TrafficCongestionProvider ->
             val extractedRouteDataItems = extractRouteData(
```

Comment:
`extractRouteDataWithTrafficAndRoadClassDeDuped` and `extractedRouteDataItems` can be merge into one now.

Reply:
I don't think they can be.  When the soft gradient option is active the route data can't be de-duped. If it is the soft gradient won't appear correctly.